### ISSUE TITLE
Feature request: Will you open-source the ranking algorithm of Matters?

### ISSUE THREAD

**Issue by askender** [thematters/developer-resource#11]
Very curious!

**Comment by guoliu** [comment]
Yes, we are planning to open source the entire codebase, likely within 2 weeks.

**Comment by guoliu** [comment]
@askender sorry things took longer than we expected, but here is our [open source plan](https://matters.news/@hi176/matters%E9%96%8B%E6%BA%90%E8%A8%88%E7%95%AB%E5%95%9F%E5%8B%95-%E9%82%80%E8%AB%8B%E4%BD%A0%E4%BE%86%E5%85%B1%E5%BB%BA%E9%A6%AC%E7%89%B9%E5%B8%82-bafyreiadaxjjwxk6mhsx4u6ognhxw3atlwiloscyn7r2iff6tsz4rrc4by) we just launched. Eventually all repos will be open sourced, but we are inviting experienced developer first to discuss workflows and coordination with non-tech users. If you are interested you are more than welcome to join and take a look at the ranking methods we are using now 😄

**Comment by guoliu** [comment]
@askender The resolvers of recommendations can be found here: https://github.com/thematters/matters-server/tree/develop/src/queries/user/recommendation Most article rankings are in materialized views, which is created via database migration: https://github.com/thematters/matters-server/tree/develop/db/migrations 

For example `recommendation.valued` is the API for the main feed, and the corresponding materialized view is `article_value_materialized`, which was updated here: https://github.com/thematters/matters-server/blob/develop/db/migrations/20200720193816_update_article_value_view.js

Does this answer your question?
